atc/atccmd/config_types.go

```diff
@@ -0,0 +1,208 @@
+package atccmd
+
+import (
+	"net"
+	"time"
+
+	"github.com/concourse/concourse/atc/creds"
+	"github.com/concourse/concourse/atc/creds/conjur"
+	"github.com/concourse/concourse/atc/creds/credhub"
+	"github.com/concourse/concourse/atc/creds/dummy"
+	"github.com/concourse/concourse/atc/creds/kubernetes"
+	"github.com/concourse/concourse/atc/creds/secretsmanager"
+	"github.com/concourse/concourse/atc/creds/ssm"
+	"github.com/concourse/concourse/atc/creds/vault"
+	"github.com/concourse/concourse/atc/metric"
+	"github.com/concourse/concourse/atc/metric/emitter"
+	"github.com/concourse/concourse/atc/policy"
+	"github.com/concourse/concourse/atc/worker"
+	"github.com/concourse/concourse/skymarshal/skycmd"
+	"github.com/concourse/flag"
+)
+
+type TLSConfig struct {
+	BindPort uint16    `yaml:"bind_port,omitempty"`
+	Cert     flag.File `yaml:"cert,omitempty"`
+	Key      flag.File `yaml:"key,omitempty"`
+	CaCert   flag.File `yaml:"ca_cert,omitempty"`
+}
+
+// IMPORTANT! The env tags are only added for backwards compatibility sake. Any
+// new fields do NOT require the env tag
+type LetsEncryptConfig struct {
+	Enable  bool     `yaml:"enable,omitempty" env:"CONCOURSE_ENABLE_LETS_ENCRYPT,CONCOURSE_LETS_ENCRYPT_ENABLE"`
```

Comment:
If you are wondering why some struct values have `env` and some don't, its because I use my library called `envstruct` and it will build up an environment variable name based on the tags within the struct. For example, this `LetsEncryptConfig` is actually within a struct

```
type RunConfig struct {
  LetsEncrypt LetsEncryptConfig `yaml:"lets_encrypt"`
}
```

So the env that will be built up for the `Enable` field is `CONCOURSE_LETS_ENCRYPT_ENABLE` because it appends the prefix (which is set to `CONCOURSE`) to the `yaml` tag in the nested struct for `LetsEncrypt` (which is `lets_encrypt` but uppercased) and then to the value of the `yaml` tag for `Enable` (which is `enable` but uppercased). When we try to find an environment variable for this field, it will use that built up string `CONCOURSE_LETS_ENCRYPT_ENABLE`.

The problem is that fetching env is purely for backwards compatibility purposes and the env that we used before this config PR was `CONCOURSE_ENABLE_LETS_ENCRYPT` which is different than `CONCOURSE_LETS_ENCRYPT_ENABLE`. So setting a tag value for `env` will override the env variable that is fetched from `envstruct`.